tempodb/backend/local/local.go

```diff
@@ -131,6 +131,7 @@ func (rw *Backend) Read(ctx context.Context, name string, keypath backend.KeyPat
 
 	stat, err := f.Stat()
 	if err != nil {
+		f.Close()
```

Comment:
I am not sure about this one - should we close if open succeeded and stat failed?

Reply:
how about just deferring close? i think that's ok for read only files.

Reply:
Cant defer close in the higher scope because f is returned from the function. I assume the caller expects an opened file